health_monitor/health_monitor.ino

```diff
@@ -0,0 +1,167 @@
+}
+
+BLEScanResults scan_ble()
+{
+    auto *scan = BLEDevice::getScan();
+    scan->setActiveScan(true);
+    scan->start(SCAN_SECONDS);
+    delay(1000 * SCAN_SECONDS);
+    scan->stop();
+    return scan->getResults();
+}
+
+unsigned int count_infected(BLEScanResults &results)
+{
+    unsigned int infected = 0;
+    for (auto i = 0; i < results.getCount(); ++i)
+    {
+        auto device = results.getDevice(i);
+        auto rssi = device.getRSSI();
+        auto name = device.getName();
+
+        if (is_monitor(name) && 
+            is_close(rssi) && 
+            is_infected(name)) {
+            ++infected;
+        }
+    }
+    return infected;
+}
+
+void print_scan_results(BLEScanResults &results)
+{
+    Serial.println("---<< scan results >>--");
+    for (auto i = 0; i < results.getCount(); ++i)
+    {
+        auto device = results.getDevice(i);
+        auto address = BLEAddress(device.getAddress());
+        auto rssi = device.getRSSI();
+        auto name = device.getName();
+
+        Serial.print("Addr: ");
+        Serial.print(address.toString().c_str());
+        Serial.print(", RSSI: ");
+        Serial.print(rssi);
+        Serial.print(", Name: ");
+        Serial.print(name.c_str());
+        Serial.println("");
+    }
+    Serial.println("---");
+}
+
+std::string to_state(unsigned int health) {
+    if (health == 0)
+    {
+        return IMMUNE;
+    }
+    else if (health < 10)
+    {
+        return SUPER;
+    }
+    else if (health < 50)
+    {
+        return HEALTY;
+    }
+    else if (health < 60)
+    {
+        return A_SYMPTOMATIC;
+    }
+    else if (health < 99)
+    {
+        return SICK;
+    }
+    return ZOMBIE;
+}
+
+unsigned int update_health(
+    unsigned int current_health,
+    unsigned int infected)
+{
+    if (health == 0 || health == 100) {
+        return current_health;
+    }
+
+    if (infected > 0) {
+        return current_health + 1;
+    }
+
+    return current_health;
+}
+
+void setup()
+{
+    // start serial
+    Serial.begin(115200);
+
+    // Increment boot number and print it every reboot
+    ++bootCount;
+    Serial.println("Boot number: " + String(bootCount));
+
+    // String representing our state (used by other devices to observe us)
+    auto state = to_state(health);
+    Serial.println(state.c_str());
+    Serial.println("Health: " + String(health));
+
+    // Start bluetooth to advertise our state
+    BLEDevice::init(PREFIX + state);
+    BLEDevice::startAdvertising();
+
+    // Scan for nearby devices and count infected
+    auto devices = scan_ble();
+    print_scan_results(devices);
+    auto infected = count_infected(devices);
+    Serial.println("Infected count: " + String(infected));
+
+    // Update our health value
+    health = update_health(health, infected);
+
+    // Flush the serial buffer
+    Serial.flush();
+
+    // Enable waking up in some amount of time and sleep
+    esp_sleep_enable_timer_wakeup(random(1, TIME_TO_SLEEP) * uS_TO_S_FACTOR);
```

Comment:
Might be worth double checking if `random` returns the same value after deep-sleep reset. I'm not sure if, without an explicit call to `randomSeed(/*some random seed source*/)`, it will produce different values after being set back to its default state.

Reply:
Ok, I just confirmed that the same number is not always produced everytime